src/android/com/mady/wifi/api/WifiHotSpots.java

```diff
@@ -247,6 +250,33 @@ public boolean connectToHotspot(String netSSID, String netPass, Integer authenti
         return false;
     }
 
+    /**
+     * Determines if current connection is a captive portal by checking if we get a 204 response
+     * from google. If we do not, we can assume this is do to a captive portal instead returning a 200 response.
+     * Logic taken from https://stackoverflow.com/questions/13958614/how-to-check-for-unrestricted-internet-access-captive-portal-detection
+     */
+    public boolean isCaptivePortalConnection() {
+        HttpURLConnection urlConnection = null;
+        try {
+            URL url = new URL("http://clients3.google.com/generate_204");
```

Comment:
```suggestion
            URL url = new URL("https://clients3.google.com/generate_204");
```

Since this is a Google URL, and google just *loves* https, it might be a good idea to use https, just in case they decide to drop http for this route.

I also found that routes https://clients[1-6].google.com/generate_204 are also valid. It might be a good idea, if one of them fails, to try another.

Reply:
I just tried using those https routes and they all threw the error `exception javax.net.ssl.SSLException: Connection closed by peer` when trying to check for the 204 response code. I'm assuming it was failing to make an active connection with these routes. It's only working with the http routes so going to leave as is.

Reply:
Those routes work fine from my computer. You shouldn't be getting SSL errors. That indicates some other problem on your end.

Reply:
It's the nature of the captive portal network before the user is authenticated, it can only connect to `http` addresses. Happens on my computer when connected to the captive portal network before authenticating-
![image](https://user-images.githubusercontent.com/45437768/69731388-6ff86780-10ef-11ea-85ec-d1347c8f5893.png)

Reply:
Whereas accessing an `http` address triggers the redirect to the captive portal login

Reply:
Makes sense.